custom_components/kumo/config_flow.py

```diff
@@ -58,14 +62,46 @@ async def async_step_user(self, user_input=None):
         if user_input is not None:
             try:
                 info = await validate_input(self.hass, user_input)
-                return self.async_create_entry(
-                    title=info["title"],
-                    data={
-                        "username": user_input["username"],
-                        "password": user_input["password"],
-                        "prefer_cache": user_input["prefer_cache"],
-                    },
+
+                account = KumoCloudAccount(
+                    user_input["username"], user_input["password"]
+                )
+                await self.hass.async_add_executor_job(account.try_setup)
+                self.kumo_cache = await self.hass.async_add_executor_job(
+                    account.get_raw_json
                 )
+                self.user_account_setup = user_input
+                self.title = info["title"]
+                self.units = []
+                for child in self.kumo_cache[2]["children"]:
+                    for raw_unit in child["zoneTable"].values():
```

Comment:
Also need to go through the "children" as in [pykumo](https://github.com/dlarrick/pykumo/blob/master/pykumo/pykumo.py#L537-L542) in order to handle units that are in changeover groups.